Content.Server/UserInterface/ActivatableUISystem.cs

```diff
@@ -85,17 +86,33 @@ private void AddOpenUiVerb(EntityUid uid, ActivatableUIComponent component, GetV
     private void OnActivate(EntityUid uid, ActivatableUIComponent component, ActivateInWorldEvent args)
     {
         if (args.Handled) return;
+        if (component.AllowedItems != null) return;
         if (component.InHandsOnly) return;
         args.Handled = InteractUI(args.User, component);
     }
 
     private void OnUseInHand(EntityUid uid, ActivatableUIComponent component, UseInHandEvent args)
     {
         if (args.Handled) return;
+        if (component.AllowedItems != null) return;
         if (component.rightClickOnly) return;
         args.Handled = InteractUI(args.User, component);
     }
 
+    private void OnInteractUsing(EntityUid uid, ActivatableUIComponent component, InteractUsingEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (component.AllowedItems == null)
+            return;
+
+        if (component.AllowedItems.IsValid(args.Used, EntityManager))
```

Comment:
wouldnt it be better to move these checks to InteractUI and following the same pattern in here as the ones in OnUseInHand and OnActivate?

Reply:
There is a problem with this: `InteractUI` can't check what item was used, and if can't even check whether it was called by item or by hand. 
So, we either add bool `UseOnlyByItems` and pass this item to the `InteractUI` or keep it as it is. I can't say which is better.